omod/src/main/webapp/resources/scripts/visit/filters.js

```diff
@@ -494,12 +498,12 @@ angular.module("filters", [ "uicommons.filters", "constants", "encounterTypeConf
 
                 if (it.require) {
                     // find a cleaner way to avoid polluting scope
-                    var result = (function(expr) {
+                    var result = (function(expr, visit) {
                         var sessionLocation = SessionInfo.get().sessionLocation;
                         var user = new OpenMRS.UserModel(SessionInfo.get().user);
                         var patient = visit ? visit.patient : null;
```

Comment:
It appears that `visit` is already present in the closure. Why are you passing it as a parameter?